File: python/hot_100/11_container-with-most-water.py
from typing import List
class Solution:
    def maxArea(self, height: List[int]) -> int:
        area=0
        i=0
        j=len(height)-1
        while i<j:
            area=max(area,min(height[i],height[j])*(j-i))
            if height[i]<height[j]:
                i+=1
            else:
                j-=1
        return area

File: go/hot_100/11_container-with-most-water.go
package main

func maxArea(height []int) int {
	l, r := 0, len(height)-1
	maxA := 0
	for l != r {
		if height[l] <= height[r] {
			maxA = Max(maxA, height[l]*(r-l))
			l += 1
		} else {
			maxA = Max(maxA, height[r]*(r-l))
			r -= 1
		}
	}
	return maxA
}

// func main() {
// 	maxArea([]int{1, 8, 6, 2, 5, 4, 8, 3, 7})
// }
